### PR TITLE
introduce V2 threads

### PR DESCRIPTION
**Pull request by BalzGuenat**
This addresses the concerns raised in issue https://github.com/BalzGuenat/CustomThreads/issues/2 and PR https://github.com/BalzGuenat/CustomThreads/pull/3.

The changes bring the generated thread profiles closer to standard metric profiles. In particular, it removes the previously fixed tolerances which caused problems with certain thread parameters. This also means that now the `O.0` tolerance class is actually a zero tolerance fit.

Of course, these new threads are incompatible with those generated previously. In order not to break people's models, the [3DPrintedMetric.xml](https://github.com/BalzGuenat/CustomThreads/blob/master/3DPrintedMetric.xml) file remains unchanged. The threads are now provided in a new file `3DPrintedMetricV2.xml` and under a new name "3D-printed Metric Threads V2". People who don't have models with old threads only need this file.

I would kindly ask @zhogov and @jil24 to test these changes and report if they fix their respective issues.

### CODE REVIEW

**Comment by zhogov** [comment]
Here's how threads look in v2 compared to master and #3:
image
this is 24x3.5 offset 0.1

I like how they look now, thank you so much! 

Nitpick: two versions in the middle have "min" and "max" vertical lines have same length and this 
- looks aesthetically pleasing
- would make threads sturdier as thread has more balanced distribution of material between the "crest" and "flank" parts
- corresponds more to wiki page for ISO threads:
400px-ISO_and_UTS_Thread_Dimensions svg

File: main.py [hunk 1]
```diff
@@ -57,63 +55,71 @@ def sizes(self):
...
+                ET.SubElement(thread_element, "Gender").text = thread.gender
+                ET.SubElement(thread_element, "Class").text = thread.clazz
+                ET.SubElement(thread_element, "MajorDia").text = "{:.4g}".format(thread.majorDia)
+                ET.SubElement(thread_element, "PitchDia").text = "{:.4g}".format(thread.pitchDia)
+                ET.SubElement(thread_element, "MinorDia").text = "{:.4g}".format(thread.minorDia)
+                if thread.tapDrill:
+                    ET.SubElement(thread_element, "TapDrill").text = "{:.4g}".format(thread.tapDrill)
+
+    ET.indent(tree)
+    tree.write('output.xml', encoding='UTF-8', xml_declaration=True)
```
Comment (zhogov) [inline main.py:122]:
```suggestion
    tree.write('3DPrintedMetricV2.xml', encoding='UTF-8', xml_declaration=True)
```

**Comment by BalzGuenat** [comment]
>     * corresponds more to wiki page for ISO threads:
>       400px-ISO_and_UTS_Thread_Dimensions svg

You will notice the ISO profiles also have different lengths of "vertical lines" (or in the above diagram horizontal lines), so I would consider the new threads more corresponding to ISO.

**Comment by zhogov** [comment]
> > ```
> > * corresponds more to wiki page for ISO threads:
> >   400px-ISO_and_UTS_Thread_Dimensions svg
> > ```
> 
> You will notice the ISO profiles also have different lengths of "vertical lines" (or in the above diagram horizontal lines), so I would consider the new threads more corresponding to ISO.

looked closer – true, they do indeed :sweat_smile:

**Review by jil24 — APPROVED** [review]
This looks great. I can only review the changes as I won’t have an opportunity to test it for some time.

**Pull request merged**